File: p007/p007.py
def nthPrime(n):
    primeList = []
    i = 2
    j = 1
    while j <= n:
        isPrime = True
        for k in primeList:
            if i % k == 0:
                isPrime = False
                break
        if isPrime:
            primeList.append(i)
            j += 1
        i += 1
    return primeList[-1]

if __name__ == "__main__":
    print(nthPrime(10001))

File: p007/p007.js
function nthPrime(n) {
  // Good luck!
  const primeList = [];
  let i = 2,
    j = 1;
  while (j <= n) {
    let isPrime = true;
    for (let k = 0; k < primeList.length; k++) {
      if (i % primeList[k] == 0) {
        isPrime = false;
        break;
      }
    }
    if (isPrime) {
      primeList.push(i);
      j += 1;
    }
    i += 1;
  }
  return primeList[primeList.length - 1];
}

nthPrime(10001);
